### PR TITLE
Add `gap` properties to `replace-spacing-lengths` migration

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?

Part of #7212  

### WHAT is this pull request doing?

Adding more functionality to the `replace-spacing-lengths` migration script.

Running ```npx @shopify/polaris-migrator replace-spacing-lengths "**/*.scss"``` will now also target the following `gap` properties and replace any `px`, `rem` or `rem()` values that align with our [spacing tokens](https://github.com/Shopify/polaris/blob/a192e3641a8629a2ea9e1a1bd5d6c20aaf9a499c/polaris-tokens/src/token-groups/spacing.ts). 

- `gap`
- `grid-gap`
- `row-gap`
- `grid-row-gap`
- `column-gap`
- `grid-column-gap`

Co-Authored-By: Chloe Rice <18447883+chloerice@users.noreply.github.com>

### CODE REVIEW

File: .changeset/dirty-students-move.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-migrator': minor
+---
+
+Add `gap` and `border` properties to `replace-sass-lengths` migration
```
Comment (lgriffee) [inline .changeset/dirty-students-move.md]:
```suggestion
Rename `replace-sass-length` migration to `replace-sass-space`
Add `gap` properties to `replace-sass-space` migration
```

**Review by aaronccasanova — APPROVED** [review]

**Review by sam-b-rose — APPROVED** [review]
So clean!! Looks great💯

File: polaris-migrator/README.md [hunk 2]
```diff
@@ -49,6 +49,25 @@ A generic codemod to rename any component prop.
 npx @shopify/polaris-migrator rename-component-prop <path> --component=MyComponent --from=prop --to=newProp
 ```
 
+### `replace-spacing-lengths`
+
+Replace length values (`px`, `rem` and `rem()`) in spacing declarations (`padding`, `margin`, and `gap`) with the corresponding Polaris spacing token.
```
Comment (lgriffee) [inline polaris-migrator/README.md]:
```suggestion
Replace lengths (`px`, `rem` and `rem()`) in spacing declarations (`padding`, `margin`, and `gap`) with the corresponding Polaris spacing token.
```

File: polaris-migrator/README.md [hunk 3]
```diff
@@ -49,6 +49,25 @@ A generic codemod to rename any component prop.
 npx @shopify/polaris-migrator rename-component-prop <path> --component=MyComponent --from=prop --to=newProp
 ```
 
+### `replace-spacing-lengths`
+
+Replace lengths (`px`, `rem` and `rem()`) in spacing declarations (`padding`, `margin`, and `gap`) with the corresponding Polaris spacing token.
```
Comment (lgriffee) [inline polaris-migrator/README.md]:
```suggestion
Replace lengths and functions (`px`, `rem` and `rem()`) in spacing declarations (`padding`, `margin`, and `gap`) with the corresponding Polaris spacing token.
```

**Pull request merged**